### PR TITLE
Feature cleanup memory leaks

### PR DESCRIPTION
**Pull request by Nielsbishere**
https://github.com/TeamWisp/WispRenderer/issues/387

### CODE REVIEW

File: src/frame_graph/frame_graph.hpp [hunk 1]
```diff
@@ -334,9 +334,9 @@ namespace wr
 			}
 
 			// Make sure we free the data objects we allocated.
-			for (auto& data : m_data)
+			for (void *data : m_data)
 			{
-				delete data;
+				free(data);
```
[the remark this answers is not in the document]
Comment (Nielsbishere) [reply src/frame_graph/frame_graph.hpp]: No.

[the remark this answers is not in the document]
Comment (Nielsbishere) [reply src/frame_graph/frame_graph.hpp]: Okay maybe.

[the remark this answers is not in the document]
Comment (Nielsbishere) [reply src/frame_graph/frame_graph.hpp]: I'll change it to std::shared_ptr<void>like suggested

**Comment by VZout** [comment]
not mergable

**Pull request closed without merging**